src/rhsm/connection.py

```diff
@@ -762,54 +841,58 @@ def _set_accept_language_in_header(self):
-        handler = self.apihandler + method
-
-        # We try to import it here to get fresh value, because rhsm.service can receive
-        # several D-BUS API calls with different locale argument (every request have to have
-        # different locale)
-        self._set_accept_language_in_header()
-
-        # Load certificates from cert dir if specified
-        if cert_key_pairs is None or len(cert_key_pairs) == 0:
-            cert_key_pairs = self._get_cert_key_list()
-
-        if (
-            headers is not None
-            and "Content-type" in headers
-            and headers["Content-type"] == "application/x-www-form-urlencoded"
-        ):
-            body = urlencode(info).encode("utf-8")
-        elif info is not None:
-            body = json.dumps(info, default=json.encode)
-        else:
-            body = None
-
-        log.debug("Making request: %s %s" % (request_type, handler))
-
-        if self.user_agent:
-            self.headers["User-Agent"] = self.user_agent
-
-        final_headers = self.headers.copy()
-        if body is None:
-            final_headers["Content-Length"] = "0"
-        if headers:
-            final_headers.update(headers)
-
-        self._print_debug_info_about_request(request_type, handler, final_headers, body)
+    @staticmethod
+    def parse_keep_alive_header(keep_alive_header: str) -> tuple:
+        """
+        Try to parse 'Keep-Alive' header received from candlepin server
+        :param keep_alive_header: string with value of the header
+        :return: Tuple containing connection timeout and maximal number of requests
+        """
+        keep_alive_timeout = None
+        max_request_num = None
+        pattern = re.compile(r"^(.*)=(\d+)$")
+
+        items = keep_alive_header.split()
+
+        for item in items:
+            # Timeout of connection using keep-alive
+            if item.startswith("timeout=") is True:
+                search_result = pattern.search(item)
+                if search_result is not None:
+                    keep_alive_timeout = search_result.groups()[1]
+                    if keep_alive_timeout.isdecimal() is True:
+                        keep_alive_timeout = int(keep_alive_timeout)
+                    else:
+                        keep_alive_timeout = None
+            # Maximal number of request on one connection
+            if item.startswith("max=") is True:
+                search_result = pattern.search(item)
+                if search_result is not None:
+                    max_request_num = search_result.groups()[1]
+                    if max_request_num.isdecimal() is True:
+                        max_request_num = int(max_request_num)
+                    else:
+                        max_request_num = None
+
+        return keep_alive_timeout, max_request_num
+
+    def _make_request(self, request_type, handler, final_headers, body, cert_key_pairs):
+        """
 
+        :return:
+        """
```

Comment:
still undocumented, i'd remove it then (i don't see any added value by this docstring)